### ISSUE TITLE
readonly attribute on input type number is not announced by NVDA

### ISSUE THREAD

**Issue by Sommerli2** [nvaccess/nvda#20525]
### Brief summary

NVDA fails to announce the read only state for native HTML <input type = "number" readonly> elements when they receive focus. While the readonly attribute is correctly announced on standard text inputs, it is completely ignored on number inputs across all major browsers, leaving screen reader users unaware that the value cannot be modified. 

### Steps to reproduce

1. Open a web page gontaining a native HTML5 number input with the readonly attribute. For example: https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/input/number 

code example: <input type="number" id="number" name="number" readonly/>
2. Tab to the input element and listen to the NVDA

### Actual behavior

NVDA announces the input field and its current value, but completely ignores the readonly state. There is no verbal indication thta the field's value vannot be modiefied by the user. 

### Expected behavior

NVDA announce "read only" when the element receives focus, identical to how it handles input type "text" with a readonly attribute. 

### NVDA logs, crash dumps and other attachments

None required as this is a purely semantic reproduction issue. 

### NVDA type

installed copy

### NVDA version

2025.3

### Windows version

Windows 11

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I do not have any NVDA add-ons installed

### Does the issue still occur after you run the System Accessibility Repair Tool (COM Registration Fixing Tool in older versions) in NVDA's tools menu?

I have not run the System Accessibility Repair Tool

**Comment by Sommerli2** [comment]
same problem on input type date.

**Comment by DVZteama11y** [comment]
The bug also affects version 2026.1.1. 
The attribute `aria-readonly=‘true’` is not announced either.
